neural_modelling/src/neuron/spike_processing.c

```diff
@@ -442,36 +418,26 @@ void user_event_callback(uint unused0, uint unused1) {
 /* INTERFACE FUNCTIONS - cannot be static */
 
 //! \brief clears the input buffer of packets and records them
-int spike_processing_clear_input_buffer(void) {
+void spike_processing_clear_input_buffer(timer_t time) {
 
-    // Record the number of packets received this timer tick
-    int to_return = packets_this_time_step;
-    packets_this_time_step = 0;
+    // Record the number of packets received last timer tick
+    p_per_ts_struct.time = time;
+    recording_record(p_per_ts_region, &p_per_ts_struct, sizeof(p_per_ts_struct));
```

Comment:
this is what i wanted to avoid. i wanted to use the neuron recording code here. but some form of c make issue occurred here. id want neuron recording so that we could do sampling intervals and so that time was just updated in  the same way.

Reply:
also meant i could record other types this way etc.

Reply:
You can record any type you want here.  Just pass the type.  Note that the neuron recording also would require this if the type you want doesn't happen to have a function for it.

Reply:
yes, but the neuron recording had a type for this. but im getting closer and closer to just agreeing with ya.......

Reply:
ah bah. giving up